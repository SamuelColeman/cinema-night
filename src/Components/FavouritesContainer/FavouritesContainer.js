import React from 'react';
import FavouriteCard from '../FavouriteCard/FavouriteCard';
import './FavouritesContainer.css';
import { connect } from 'react-redux';

const FavouritesContainer = ({ favouritesList, handleFavourite }) => {
	console.log(favouritesList.favorites)
	let mapFavorites;
	if (favouritesList.favorites === undefined) {
		mapFavorites = <h1>Error: Login</h1>
	} else { 
		mapFavorites = favouritesList.favorites.map(movie => {
		return <FavouriteCard key={movie.id} movie={movie} handleFavourite={handleFavourite} />
	})
}
	return (
		<section>
			<h1>Favourites:</h1>
			{mapFavorites}
		</section>
	)
}

export const mapStateToProps = (state) => ({
  favouritesList: state.favouritesList
})

export default connect(mapStateToProps)(FavouritesContainer);