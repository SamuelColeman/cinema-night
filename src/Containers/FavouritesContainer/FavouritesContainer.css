.favorites-card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.button-back {
  max-width: 4%;
  border-radius: 40px;
  float: left;
  margin-left: 3%;
  margin-top: 2.5%;


}

.button-back:hover {
  max-width: 5%
}

.favcontain__h2--title{
  border-bottom:solid;
}