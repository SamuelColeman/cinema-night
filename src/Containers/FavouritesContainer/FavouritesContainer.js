import React from 'react';
import FavouriteCard from '../FavouriteCard/FavouriteCard';
import './FavouritesContainer.css';
import { connect } from 'react-redux';
import back from '../../images/back.png'
import { Link } from 'react-router-dom';

export const FavouritesContainer = ({ favouritesList, handleFavourite }) => {
	let mapFavorites;
	if (favouritesList.favorites === undefined) {
		mapFavorites = <h1>Error: Login</h1>
	} else { 
		mapFavorites = favouritesList.favorites.map(movie => {
		return <FavouriteCard key={movie.id} movie={movie} handleFavourite={handleFavourite} />
	})
}
	return (
		<section>
			<Link to='/'>
			<img className="button-back" src={back}></img>
			</Link>
			<h2>Favorites</h2>
			<div className="favorites-card-container">
			{mapFavorites}
			</div>
		</section>
	)
}

export const mapStateToProps = (state) => ({
  favouritesList: state.favouritesList
})

export default connect(mapStateToProps)(FavouritesContainer);