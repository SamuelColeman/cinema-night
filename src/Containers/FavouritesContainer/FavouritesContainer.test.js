import React from 'react';
 import { shallow } from 'enzyme';
 import { FavouritesContainer, mapStateToProps } from './FavouritesContainer';


describe('Favourites Container', () => {
    let wrapper;
    let mockHandleFavourite = jest.fn();
    let mockFavouritesList = [
        {
          "poster_path": "/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg",
          "movie_id": 420809,
          "title": "Maleficent: Mistress of Evil",
          "vote_average": 7.1,
          "overview": "Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.",
          "release_date": "2019-10-18"
        },
        {
          "poster_path": "/gibberish.jpg",
          "movie_id": 222222,
          "title": "A title",
          "vote_average": 11,
          "overview": "It's cool.",
          "release_date": "2019-10-18"
        }
    ];

    beforeEach(() => {
      wrapper = shallow( <FavouritesContainer 
        handleFavourite={mockHandleFavourite}
        favouritesList={mockFavouritesList}
      />)
    })

    it('should match snapshot with all data being passed in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
 })

