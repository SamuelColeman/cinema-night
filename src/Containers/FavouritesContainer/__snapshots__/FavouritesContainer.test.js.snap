// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites Container should match snapshot with all data being passed in correctly 1`] = `
<section>
  <h1>
    Error: Login
  </h1>
</section>
`;
